android/src/main/java/com/reactnativestripesdk/PaymentLauncherFragment.kt

```diff
@@ -70,16 +70,8 @@ class PaymentLauncherFragment(
             promise.resolve(paymentIntent)
           }
           StripeIntent.Status.RequiresAction -> {
-            if (isPaymentIntentNextActionVoucherBased(result.nextActionType)) {
-              val paymentIntent = createResult("paymentIntent", mapFromPaymentIntentResult(result))
-              promise.resolve(paymentIntent)
-            } else {
-              (result.lastPaymentError)?.let {
-                promise.resolve(createError(ConfirmPaymentErrorType.Canceled.toString(), it))
-              } ?: run {
-                promise.resolve(createError(ConfirmPaymentErrorType.Canceled.toString(), "The payment has been canceled"))
-              }
-            }
```

Comment:
I'm not sure why this was returning a `Canceled` error before, but that seems wrong to me. We're now returning a map containing `nextAction` data, so I hope this is safe but unfortunately don't have context on it (no commit history on this)

Reply:
I don't know what this is supposed to do either, and your change seems to make sense. In the iOS SDK, we don't appear to have any logic to handle RequiresAction + a voucher-based payment method differently.

Reply:
I don't have a lot of context on this either, but on Android and iOS we have the same thing: https://github.com/stripe/stripe-android/blob/master/payments-core/src/main/java/com/stripe/android/StripeIntentResult.kt#L34

We return success based on this check here: https://github.com/stripe/stripe-android/blob/master/payments-core/src/main/java/com/stripe/android/StripeIntentResult.kt#L71

I assume that we return canceled when we are in an invalid state for the payment intent type. Should we leave it as it was? I am not sure how we can verify this.

Reply:
@jameswoo-stripe that link is super helpful, looks like we're making the same check on iOS: https://github.com/stripe-ios/stripe-ios/blob/private/Stripe/STPPaymentHandler.swift#L201, and is set up with the same true/false cases. The solution here is to move from `isPaymentIntentNextActionVoucherBased` to something like `isPaymentIntentNextActionVoucherBased` to `isNextActionSuccessState`